apps/vmq_server/src/vmq_mqtt_fsm.erl

```diff
@@ -539,18 +539,20 @@ check_user(#mqtt_connect{username=User, password=Password} = F, State) ->
         false ->
             case auth_on_register(User, Password, State) of
                 {ok, QueueOpts, #state{peer=Peer, subscriber_id=SubscriberId, clean_session=CleanSession,
-                                       cap_settings=CAPSettings, def_opts=DOpts} = NewState} ->
+                                       cap_settings=CAPSettings, def_opts=DOpts} = State1} ->
                     CoordinateRegs = maps:get(coordinate_registrations, DOpts, ?COORDINATE_REGISTRATIONS),
                     case vmq_reg:register_subscriber(CAPSettings#cap_settings.allow_register, CoordinateRegs, SubscriberId, CleanSession, QueueOpts) of
-                        {ok, SessionPresent, QPid} ->
+                        {ok, #{session_present := SessionPresent,
+                               initial_msg_id := MsgId}, QPid} ->
```

Comment:
We could return the `QPid` also as part of the map.

Reply:
yeah, I guess that makes sense. Then spec it as a mandatory field in the map.